common/src/main/java/org/jboss/hal/testsuite/treefinder/TreeNavigationGroup.java

```diff
@@ -0,0 +1,126 @@
+package org.jboss.hal.testsuite.treefinder;
+
+import org.jboss.arquillian.graphene.findby.ByJQuery;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NotFoundException;
+import org.openqa.selenium.WebElement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a group of navigation items. Be aware, that containment of this group varies from several subtrees as its
+ * children to just one child with no subtree.
+ */
+public class TreeNavigationGroup {
+
+    private static final Logger logger = LoggerFactory.getLogger(TreeNavigationGroup.class);
+
+    private static final String TREE_ITEM_CLASS_NAME = "gwt-TreeItem";
+
+    private static final By DIRECT_CHILDREN_SELECTOR = ByJQuery.selector("> div > div");
+
+    private final WebElement root;
+
+    private TreeNavigationGroup() {
+        throw new UnsupportedOperationException("It is not allowed to use this constructor!");
+    }
```

Comment:
The implicit is used only if there is no constructor defined in the class. In this case, you have constructor with one param => implicit constructor isn't available.